velox/dwio/parquet/reader/ParquetData.cpp

```diff
@@ -143,4 +212,42 @@ std::pair<int64_t, int64_t> ParquetData::getRowGroupRegion(
   return {fileOffset, length};
 }
 
+std::unique_ptr<BloomFilter> getBloomFilter(
+    dwio::common::BufferedInput& bufferedInput,
+    const std::vector<thrift::RowGroup>& rowGroups,
+    const uint32_t rowGroupId,
+    const uint32_t column,
+    memory::MemoryPool& pool) {
+  if (rowGroupId < 0 || rowGroupId >= rowGroups.size()) {
+    VELOX_FAIL("Invalid row group ordinal: ", rowGroupId);
+  }
+  auto rowGroupMetadata = rowGroups[rowGroupId];
+  if (column < 0 || column >= rowGroupMetadata.columns.size()) {
+    VELOX_FAIL("Invalid column index at column ordinal ", column);
+  }
+
+  auto colChunk = rowGroupMetadata.columns[column];
+  if (colChunk.__isset.crypto_metadata) {
+    VELOX_NYI("Cannot read encrypted bloom filter yet");
+  }
+
+  if (!colChunk.meta_data.__isset.bloom_filter_offset) {
+    return nullptr;
+  }
+  auto bloomFilterOffset = colChunk.meta_data.bloom_filter_offset;
+  auto fileSize = bufferedInput.getInputStream()->getLength();
+  if (fileSize <= bloomFilterOffset) {
+    VELOX_FAIL("file size less or equal than bloom offset");
+  }
+
+  auto inputStream = bufferedInput.read(
+      bloomFilterOffset,
+      fileSize - bloomFilterOffset,
+      dwio::common::LogType::FOOTER);
+  auto bloomFilter =
```

Comment:
Every time this function is called, it would read and deserialize the bloom filter again. I think it'd be better to make the bloom filter as a class member, and if it's read before, just return it.